src/plugins/kibana_utils/public/state_management/state_hash/state_hash.ts

```diff
@@ -32,7 +32,7 @@ export function retrieveState<State>(stateHash: string): State {
 
 export function persistState<State>(state: State): string {
   const json = JSON.stringify(state);
-  const hash = createStateHash(json);
+  const hash = createStateHash(json, hashedItemStore.getItem.bind(hashedItemStore));
```

Comment:
I am not sure about providing here a second param. We didn't do it previously.

Reply:
Exactly, it removes the hardcoded `hashedItemStore` so the function can be shared in `public` and `server` code, and lets the `public` code pass `hashedItemStore.get` in as an `existingJsonProvider` instead.